Samples/VirtualCamera/VirtualCameraManager_WinRT/VirtualCameraRegistrar.cpp

```diff
@@ -0,0 +1,141 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+
+#include "pch.h"
+#include "VirtualCameraRegistrar.h"
+#include "VirtualCameraRegistrar.g.cpp"
+#include "VirtualCameraProxy.h"
+#include "Constants.h"
+#include "Common.h"
+#include "mfidl.h"
+
+#include "winrt/Windows.Media.Devices.h"
+
+
+namespace winrt::VirtualCameraManager_WinRT::implementation
+{
+    /// <summary>
+    /// Create a virtual camera instance. 
+    /// Note that the virtual camera instance gets registered only upon calling Start().
+    /// </summary>
+    /// <param name="strMediaSourceGuid"></param>
+    /// <param name="lifetime"></param>
+    /// <param name="access"></param>
+    /// <param name="friendlyName"></param>
+    /// <param name="strHookedCameraSymbolicLink"></param>
+    /// <param name="spVirtualCamera"></param>
+    void CreateVirtualCamera(
+        _In_ winrt::hstring const& strMediaSourceGuid,
+        _In_ MFVirtualCameraLifetime lifetime,
+        _In_  MFVirtualCameraAccess access,
+        _In_ winrt::hstring const& friendlyName,
+        _In_ winrt::hstring const& strHookedCameraSymbolicLink,
+        _Inout_ wil::com_ptr_nothrow<IMFVirtualCamera>& spVirtualCamera)
```

Comment:
>_Inout_ wil::com_ptr_nothrow<IMFVirtualCamera>& spVirtualCamera [](http://example.com/codeflow?start=8&length=63)

can we just return a pointer and let called attach?

Reply:
caller can change the code to do that if they want, this is a sample

Reply:
and this wont cause any issues with consumer of this winrt object is using a different version of wil?